src/ui/public/vis/editors/default/default.js

```diff
@@ -60,6 +61,24 @@ const defaultEditor = function ($rootScope, $compile) {
             $scope.vis.dirty = false;
           };
 
+          $scope.autoApplyEnabled = false;
+          if ($scope.vis.type.editorConfig.enableAutoApply) {
+            $scope.toggleAutoApply = () => {
+              $scope.autoApplyEnabled = !$scope.autoApplyEnabled;
+            };
+
+            $scope.$watch('vis.dirty', _.debounce(() => {
+              if (!$scope.autoApplyEnabled || !$scope.vis.dirty) return;
+              $scope.stageEditableVis();
+            }, 800));
+          }
+
+          $scope.submitEditorWithKeyboard = (event) => {
+            if (event.ctrlKey && event.keyCode === 13) {
```

Comment:
Use `keyCodes` from 'ui_framework/services' for the `13`. Also we want to call `event.preventDefault()` and `event.stopPropagation()` here, since otherwise we might still cause any default behavior or allow the event to pop up further.